src/Microsoft.Health.Fhir.Shared.Api/Features/Security/SecurityProvider.cs

```diff
@@ -122,75 +126,77 @@ private void AddOAuthSecurityService(ListedCapabilityStatement statement)
             var codableConceptInfo = new CodableConceptInfo();
             security.Service.Add(codableConceptInfo);
 
-            codableConceptInfo.Coding.Add(_modelInfoProvider.Version == FhirSpecification.Stu3
-                ? Constants.RestfulSecurityServiceStu3OAuth
-                : Constants.RestfulSecurityServiceOAuth);
-
-            var openIdConfigurationUrl = $"{_securityConfiguration.Authentication.Authority}/.well-known/openid-configuration";
+            Uri authorizationEndpoint = null;
+            Uri tokenEndpoint = null;
+            Uri registrationEndpoint = null;
+            Uri managementEndpoint = null;
+            Uri revocationEndpoint = null;
+            Uri introspectionEndpoint = null;
 
-            HttpResponseMessage openIdConfigurationResponse;
-            using (HttpClient httpClient = _httpClientFactory.CreateClient())
+            if (_wellKnownConfigurationProvider.IsSmartConfigured())
             {
-                try
-                {
-                    openIdConfigurationResponse = httpClient.GetAsync(new Uri(openIdConfigurationUrl)).GetAwaiter().GetResult();
-                }
-                catch (Exception ex)
+                // Attempt to fetch the SMART configuration. This may not be configured, in which case we will fall back to the OpenID configuration.
+                GetSmartConfigurationResponse smartConfiguration = _wellKnownConfigurationProvider.GetSmartConfigurationAsync(CancellationToken.None).GetAwaiter().GetResult();
+                authorizationEndpoint = smartConfiguration?.AuthorizationEndpoint;
+                tokenEndpoint = smartConfiguration?.TokenEndpoint;
+                registrationEndpoint = smartConfiguration?.RegistrationEndpoint;
+                managementEndpoint = smartConfiguration?.ManagementEndpoint;
+                revocationEndpoint = smartConfiguration?.RevocationEndpoint;
+                introspectionEndpoint = smartConfiguration?.IntrospectionEndpoint;
+
+                if (authorizationEndpoint != null && tokenEndpoint != null)
                 {
-                    _logger.LogWarning(ex, "There was an exception while attempting to read the OpenId Configuration from \"{OpenIdConfigurationUrl}\".", openIdConfigurationUrl);
-                    throw new OpenIdConfigurationException();
+                    // The minimum requirements for SMART on FHIR have been satisfied.
+                    // Set the restful-security-service code to SMART-on-FHIR.
+                    codableConceptInfo.Coding.Add(
+                        _modelInfoProvider.Version == FhirSpecification.Stu3
+                        ? Constants.RestfulSecurityServiceStu3Smart
+                        : Constants.RestfulSecurityServiceSmart);
                 }
             }
 
-            if (openIdConfigurationResponse.IsSuccessStatusCode)
+            if (authorizationEndpoint == null && tokenEndpoint == null)
```

Comment:
Should this be an `||` vs an `&&`? Is it possible that SMART is configured but only one of these isn't null?